### PR TITLE
cmake: require CMake v3.10.2

### PR DESCRIPTION
**Pull request by tchaikov**
since we dropped the support of xenial, we now have the luxury of using
newer CMake!

- for bionic: https://packages.ubuntu.com/bionic/cmake : 3.10.2
- for CentOS7:
https://dl.fedoraproject.org/pub/epel/7/x86_64/Packages/c/ : 3.13.5

so in this change,

* bump up the required version to v3.10.2
* cleanups

Signed-off-by: Kefu Chai <kchai@redhat.com>

- [ ] References tracker ticket
- [ ] Updates documentation if necessary
- [ ] Includes tests for new functionality or reproducer for bug

### CODE REVIEW

**Comment by cbodley** [comment]
from jenkins,
> PRETTY_NAME='Ubuntu 16.04.6 LTS'
> CMake 3.10.2 or higher is required.  You are running version 3.5.1

**Comment by tchaikov** [comment]
@alfredodeza @djgalloway can we label the jenkins slave so that only bionic slaves are used for testing master? currently, we choose "huge && (centos7 || trusty) && rebootable" for testing PRs. see

https://github.com/ceph/ceph-build/blob/ba96e7971db5d57cf3ca8e11c7ee71cfd84bd2b7/ceph-pull-requests/config/definitions/ceph-pull-requests.yml#L6

**Comment by djgalloway** [comment]
> can we label the jenkins slave so that only bionic and centos7 slaves are used for testing master? currently, we choose "huge && (centos7 || trusty) && rebootable" for testing PRs.

We don't actually have any Bionic slaves.  Any Ubuntu packages are built on a Xenial slave using pbuilder.  See https://github.com/ceph/mita/blob/master/deploy/playbooks/roles/common/templates/prod_nodes.py.j2#L81-L92

Notice a `huge && trusty && rebootable` slave is using `'image_name': 'Ubuntu 16.04'`.

**Comment by alfredodeza** [comment]
what we need is to re-do the labelling and cleanup, because the current way doesn't help and it was the compromise to keep things running when things migrated from the very first jenkins iteration. Having a label that says 'bionic' when it really isn't is super confusing :(

**Comment by tchaikov** [comment]
@djgalloway @alfredodeza is it possible for us to use 'Ubuntu 18.04' for 'trusty_huge' ? or better off  defining a new type of node 'bionic'?  and then, we could, as @alfredodeza suggested, re-label the slaves and clean up the templates a little bit?

**Comment by tchaikov** [comment]
retest this please.

**Comment by tchaikov** [comment]
changelog

- s/`VERSION`/`PROJECT_VERSION`/
- add a commit to ensure cmake 3.10.2.

@cbodley could you take another look?

File: src/CMakeLists.txt [hunk 1]
```diff
@@ -114,37 +114,13 @@ endif()
...
-else()
-  set(CMAKE_CXX_STANDARD 17)
-  set(CMAKE_CXX_EXTENSIONS OFF)
-  set(CMAKE_CXX_STANDARD_REQUIRED ON)
-  set(CMAKE_C_STANDARD 99)
-  # we use `asm()` to inline assembly, so enable the GNU extension
-  set(CMAKE_C_EXTENSIONS ON)
-  set(C_STANDARD_REQUIRED ON)
-endif()
+set(CMAKE_CXX_STANDARD 17)
```
Comment (tchaikov) [inline src/CMakeLists.txt]: this helps fmt to use C++17 instead of falling back to C++11. which fails the build.

**Comment by alfredodeza** [comment]
It is going to take a good chunk of effort to have true Bionic nodes, starting with Python 3 support which none of our tooling is ready for

**Pull request merged**